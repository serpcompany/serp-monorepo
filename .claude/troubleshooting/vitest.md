https://vitest.dev/guide/common-errors.html