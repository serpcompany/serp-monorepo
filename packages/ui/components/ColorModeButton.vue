<template>
  <client-only>
    <u-button
      :icon="isDark ? 'i-lucide-moon' : 'i-lucide-sun'"
      color="neutral"
      variant="ghost"
      aria-label="Toggle color mode"
      @click="isDark = !isDark"
    />
    <template #fallback>
      <div class="size-8"></div>
    </template>
  </client-only>
</template>

<script setup lang="ts">
import { useColorMode } from '#imports';

const colorMode = useColorMode();

const isDark = computed({
  get() {
    return colorMode.value === 'dark';
  },
  set() {
    colorMode.preference = colorMode.value === 'dark' ? 'light' : 'dark';
  }
});
</script>
