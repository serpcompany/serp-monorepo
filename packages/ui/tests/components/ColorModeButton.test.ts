import { mockNuxtImport } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import ColorModeButton from '../../components/ColorModeButton.vue'
import ComponentRender from '../componentRender'

let colorMode_: string = 'dark'

describe('colorModeButton Snapshot', () => {
  // Testing both light and dark mode scenarios.
  const scenarios: [string, 'light' | 'dark'][] = [
    ['light mode', 'light'],
    ['dark mode', 'dark'],
  ]

  it.each(scenarios)('renders correctly in %s', async (desc, colorMode) => {
    colorMode_ = colorMode
    mockNuxtImport('useColorMode', () => () => colorMode_)

    // Render the component and generate a snapshot
    const html = await ComponentRender(
      `ColorModeButton ${desc}`,
      {},
      ColorModeButton,
    )
    expect(html).toMatchSnapshot()
  })
})
