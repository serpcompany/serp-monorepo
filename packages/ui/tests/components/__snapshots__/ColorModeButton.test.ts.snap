// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`colorModeButton Snapshot > renders correctly in dark mode 1`] = `
<button
  aria-label="Toggle color mode"
  class="
    aria-disabled:cursor-not-allowed
    aria-disabled:opacity-75
    dark:hover:aria-disabled:bg-transparent
    dark:hover:disabled:bg-transparent
    disabled:cursor-not-allowed
    disabled:opacity-75
    focus-visible:bg-elevated
    focus:outline-none
    font-medium
    gap-1.5
    hover:aria-disabled:bg-transparent
    hover:bg-elevated
    hover:disabled:bg-transparent
    inline-flex
    items-center
    p-1.5
    rounded-md
    text-default
    text-sm
    transition-colors
  "
  type="button"
>
  <span
    aria-hidden="true"
    class="
      i-lucide:sun
      iconify
      shrink-0
      size-5
    "
  ></span>
  <!-- v-if -->
  <!-- v-if -->
</button>
`;

exports[`colorModeButton Snapshot > renders correctly in light mode 1`] = `
<button
  aria-label="Toggle color mode"
  class="
    aria-disabled:cursor-not-allowed
    aria-disabled:opacity-75
    dark:hover:aria-disabled:bg-transparent
    dark:hover:disabled:bg-transparent
    disabled:cursor-not-allowed
    disabled:opacity-75
    focus-visible:bg-elevated
    focus:outline-none
    font-medium
    gap-1.5
    hover:aria-disabled:bg-transparent
    hover:bg-elevated
    hover:disabled:bg-transparent
    inline-flex
    items-center
    p-1.5
    rounded-md
    text-default
    text-sm
    transition-colors
  "
  type="button"
>
  <span
    aria-hidden="true"
    class="
      i-lucide:sun
      iconify
      shrink-0
      size-5
    "
  ></span>
  <!-- v-if -->
  <!-- v-if -->
</button>
`;
